src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,28 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_USER_AGE = 18;
+    private static final int MIN_USER_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("User is null");
+        }
+        if (user.getAge() == null || user.getAge() < MIN_USER_AGE) {
+            throw new RuntimeException("Invalid User age value");
+        }
+        if (user.getLogin() == null || user.getLogin().isEmpty()) {
+            throw new RuntimeException("Invalid User login value");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("There is already user with that login");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < MIN_USER_PASSWORD_LENGTH) {
+            throw new RuntimeException("Invalid User password value");
+        }
+        storageDao.add(user);
+        return user;
```

Comment:
Could be simplified as 
```suggestion
        return storageDao.add(user);
```